internal/controllers/user_devices_controller.go

```diff
@@ -133,6 +133,60 @@ func NewUserDevicesController(settings *config.Settings, dbs func() *db.ReaderWr
 	}
 }
 
+// DeviceOwnershipMiddleWare check that authenticated user owns the userDeviceID in the path or has a confirmed Eth address that owns the corresponding NFT
+func (udc *UserDevicesController) DeviceOwnershipMiddleWare(c *fiber.Ctx) error {
```

Comment:
Could we divorce this from the controller? It needs a database connection and a users client, anything else?

Reply:
added `NewMiddleware` to helpers, which accepts the database connection and users client as arguments-- this means that we're now creating the users api client inside startWebAPI so it can be passed in. Is there a reason we weren't doing this before? (I noticed that GetSharedDevices, GetMintDevice, PostMintDevice etc are all establishing a connection separately)